packages/admin-portal/src/components/pages/devs-management/devs-management.tsx

```diff
@@ -181,49 +180,38 @@ export const DevsManagement: React.FC = () => {
     },
   ]
 
-  if (!isLoading && data?.length === 0) {
-    return (
-      <React.Fragment>
-        <Helper variant="info">
-          Unfortunately, there are no results that match your search criteria, please try again
-        </Helper>
-        <Link className="text-center" to={Routes.DEV_MANAGEMENT}>
-          <Button variant="primary" type="button">
-            New Search
-          </Button>
-        </Link>
-      </React.Fragment>
-    )
-  }
-
   return (
     <ErrorBoundary>
       <Section className="mb-0">
         <H3>Developer Management</H3>
       </Section>
       <DevsManagementFilterForm filterValues={filterValues} onSearch={onSearch} />
-      {isLoading || !data ? (
-        <Loader />
-      ) : (
-        <>
-          <Section>
-            <Table scrollable={true} loading={false} data={data || []} columns={columns} />
-          </Section>
-          <Section>
-            <div>Total: {totalCount}</div>
-          </Section>
-          <Pagination onChange={onPageChange} totalCount={totalCount} pageSize={pageSize} pageNumber={pageNumber} />
-          {/* <SetDeveloperStatusModal
+      {isLoading || !data ? <Loader /> : renderResult(data, columns, totalCount)}
```

Comment:
I think you can remove `isLoading || !data` and move it to renderResult`:

`renderResult(isLoading, data, columns, totalCount)`
and then in `renderResult`
`if (isLoading || !data) {return <Loader />}`
`if (data?.length === 0){
	return <Alert message="No Results " type="info" />
}`

It's better to move logic into one place

Reply:
you can remove isLoading here, just return Loader before return component